staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -465,7 +465,7 @@ func sliceFromUnstructured(sv, dv reflect.Value, ctx *fromUnstructuredContext) e
 			}
 			dv.SetBytes(data)
 		} else {
-			dv.Set(reflect.Zero(dt))
+			dv.Set(reflect.MakeSlice(dt, 0, 0))
```

Comment:
I don't think it's correct to unconditionally make this non-nil... we should only do this if the value is non-nil, to match json marshaling, right?

https://go.dev/play/p/qFZkccUVqAh

Reply:
oh, I misread this... this is where the value is a string and we're converting back to []byte